src/Examine.RemoteDirectory/IRemoteDirectory.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Lucene.Net.Store;
+
+namespace Examine.RemoteDirectory
+{
+    public interface IRemoteDirectory
+    {
+        bool FileExists(string filename);
```

Comment:
I wonder if this is now substitutable with the Lucence Directory abstract class?

Reply:
@nzdev do we really want IRemoteDirectory to implement everything from Lucene Directory abstract class? Hmmm I am seeing there more cons then pros, as then with getting porting that to lucene 4.8 we will need refactor that class to implement changes between lucene.net and we can stay with same logic between versions :)